documentation/source/python_entry_point.rst

```diff
@@ -0,0 +1,28 @@
+Python Entry Point
+==================
+
+By default, cocotb starts the :class:`RegressionManager` as the testing framework for cocotb. However, there are many reason why a user might want to supplant the whole or part of the existing Python framework with an alternative. To this end, we have defined :envvar:`PYTHON_ENTRY`, to allow the user to enter into an alternative testing framework.
+
+.. versionadded:: 2.0
+
+How It Works
+------------
+
+After GPI implementations (VPI, VHPI, FLI, etc.) are registered, the Python implementation to the GPI (henceforth, PyGPI) starts a Python interpreter, loads the ``cocotb`` module, gets references to necessary helper functions from that module, and then calls into the ``_initialise_testbench`` function in that module to start the regression manager. The :envvar:`PYTHON_ENTRY` variable allows the user to specify a module and function other than ``cocotb._initialise_testbench`` that contains the alternative testing framework.
```

Comment:
We have slowly americanized the spelling also of functions etc. as we touched them, so maybe better ``_initialize_testbench``.

Reply:
This is a name of an existing function. Should the function be renamed in concert?

Reply:
Existing functions should probably stay the same, I didn't realize that.